xdsl/ir/core.py

```diff
@@ -478,13 +478,13 @@ def remove_use(self, use: Use):
 
 
 @dataclass(eq=False)
-class SSAValue(IRWithUses, ABC):
+class SSAValue(Generic[AttributeCovT], IRWithUses, ABC):
```

Comment:
My feeling is that this is unsound unless we make SSAValue immutable first, which blocks most of this PR stack.

Reply:
I am currently trying to see how easy it is to make this immutable, I'll probably add 1-2 commits in the stack

Reply:
https://github.com/xdslproject/xdsl/pull/4125 is making them immutable now

Reply:
Yay!